### PR TITLE
#798 Show user testimonials in the web site

### PR DESCRIPTION
**Pull request by LIUJIAHAOCS**
screen shot 2015-12-04 at 3 46 41 pm
The Testimonials are added in the index page. the long ones are selected.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -2,7 +2,10 @@
 layout: default
 title: PowerPointLabs
 ---
-
+<style>
+	.testDiv{ width:730px; height:80px}
```
Comment (kai33) [inline index.html]: 1. don't write css here. all css codes are put into main.less, so that the style can be managed in one place.
2. in order to make it maintainable, pls give it a meaningful name (like testimonial-row or testimonial-container); if this element is unique (cannot appear more than once), use id instead of class.

File: js/testimonial.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+,"Just discovered this plug-in, and i have fallen in love with it, only after the first 10 pages of the tutorial! <br>--Migauel, Spain"
+,"Thanks guys for this awesome add-in. I´m looking forward to create awesome presentations with this features that you just gave me. <br>-Jarami"
+,"I've just tried out your software, and I must say it's a excellent idea to supercharge PowerPoint presentations. <br>-- Lasith, Japan"
+,"I want to thank you for developing such cool features for PowerPoint productivity. I already contacted a bunch of friends and encourage them to try it on. Specially, the Shapes Lab utility, it is one the greatest time savers I could find. <br>-- Jimmy, Peru"
+,"Awesome plug-in! Thank you so much. This really helps give PPT some of the features that Keynote shined at. (i.e. Magic Move, etc.) This will be a fantastic timesaver! <br>-- Joe, Minneapolis"
+);
+var size = TESTIMONIALS.length;
+var i = 0;
+function showTestimonials(){
+document.getElementById("testimonialContainer").innerHTML = TESTIMONIALS[i];
```
Comment (kai33) [inline js/testimonial.js]: add indentation

File: js/testimonial.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+,"As I am a power user of PowerPoint, I truly appreciate those usable and powerful features that are provided with the plugin. <br>--Simon, Singapore"
+,"Just discovered this plug-in, and i have fallen in love with it, only after the first 10 pages of the tutorial! <br>--Migauel, Spain"
+,"Thanks guys for this awesome add-in. I´m looking forward to create awesome presentations with this features that you just gave me. <br>-Jarami"
+,"I've just tried out your software, and I must say it's a excellent idea to supercharge PowerPoint presentations. <br>-- Lasith, Japan"
+,"I want to thank you for developing such cool features for PowerPoint productivity. I already contacted a bunch of friends and encourage them to try it on. Specially, the Shapes Lab utility, it is one the greatest time savers I could find. <br>-- Jimmy, Peru"
+,"Awesome plug-in! Thank you so much. This really helps give PPT some of the features that Keynote shined at. (i.e. Magic Move, etc.) This will be a fantastic timesaver! <br>-- Joe, Minneapolis"
+);
+var size = TESTIMONIALS.length;
+var i = 0;
```
Comment (kai33) [inline js/testimonial.js]: pls avoid global variables in javascript by using anonymous function

File: index.html [hunk 4]
```diff
@@ -2,7 +2,10 @@
 layout: default
 title: PowerPointLabs
 ---
-
+<style>
+	.testDiv{ width:730px; height:80px} 
+</style>
+<body onload="showTestimonials()">
```
Comment (kai33) [inline index.html]: put and invoke this in testimonial.js; it becomes more maintainable when html, css, and js stuff is separated well

File: index.html [hunk 5]
```diff
@@ -40,8 +56,11 @@
 		<hr />
 	</div>
 
-	<div class="row col-xs-10 col-xs-offset-1 feature-grid">
+		<div class="row col-xs-10 col-xs-offset-1 feature-grid">
```
Comment (kai33) [inline index.html]: incorrect indentation here

**Comment by LIUJIAHAOCS** [comment]
screen shot 2015-12-05 at 9 45 02 am

**Comment by damithc** [comment]
The testimonial text should look different from the rest of the text. May be slightly smaller text, different font? The person name (i.e. last line) can be italic and it need not be a separate line.

**Comment by damithc** [comment]
Something like this?
image

**Comment by damithc** [comment]
Icons attached below (hard to see because they are white color), in case you need them. 
image
image

**Comment by LIUJIAHAOCS** [comment]
The icon cannot be displayed, can attach again?

**Comment by LIUJIAHAOCS** [comment]
screen shot 2015-12-05 at 12 01 18 pm
Does this font look good?

**Comment by LIUJIAHAOCS** [comment]
The icons cannot be shown, can add again?

**Comment by damithc** [comment]
Icons emailed. Don't use comic sans, ever! :-)
It's fine if it is same font. Just make it smaller and the color slightly lighter.

**Comment by LIUJIAHAOCS** [comment]
screen shot 2015-12-05 at 1 14 34 pm
How about this? Ok?

**Comment by LIUJIAHAOCS** [comment]
screen shot 2015-12-05 at 1 18 55 pm

How about this? Looks better.

**Comment by damithc** [comment]
Nice. Yes, the 2nd one is slightly better.

File: index.html [hunk 6]
```diff
@@ -25,6 +25,14 @@
 		<div class="col-xs-4">
 			<iframe width="420" height="315" src="//www.youtube.com/embed/cOmC14DGddU?rel=0" frameborder="0" allowfullscreen></iframe>
 		</div>
+		
+		<div class="row col-xs-10 col-xs-offset-1">
+					<div><img src="./img/quote-open.png" align="left"/></div>
```
Comment (kai33) [inline index.html]: incorrect indentation

File: js/testimonial.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+
+var CURRENT_TESTIMONIAL = 0;
```
Comment (kai33) [inline js/testimonial.js]: this is not a constant, pls use lower camel case and better wrap it with namespace

**Comment by kai33** [comment]
PowerPointLabs/powerpointlabs#798

File: js/testimonial.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ 	 	,"As I am a power user of PowerPoint, I truly appreciate those usable and powerful features that are provided with the plugin. &nbsp&nbsp<i>--Simon, Singapore</i>"
+ 	 	,"Just discovered this plug-in, and i have fallen in love with it, only after the first 10 pages of the tutorial! &nbsp&nbsp<i>--Migauel, Spain</i>"
+ 	 	,"Thanks guys for this awesome add-in. I´m looking forward to create awesome presentations with this features that you just gave me. &nbsp&nbsp<i>--Jarami</i>"
+ 	 	,"I've just tried out your software, and I must say it's a excellent idea to supercharge PowerPoint presentations. &nbsp&nbsp<i>-- Lasith, Japan</i>"
+ 	 	,"I want to thank you for developing such cool features for PowerPoint productivity. I already contacted a bunch of friends and encourage them to try it on. Specially, the Shapes Lab utility, it is one the greatest time savers I could find. &nbsp&nbsp<i>-- Jimmy, Peru</i>"
+ 	 	,"Awesome plug-in! Thank you so much. This really helps give PPT some of the features that Keynote shined at. (i.e. Magic Move, etc.) This will be a fantastic timesaver! &nbsp&nbsp<i>-- Joe, Minneapolis</i>"
```
Comment (damithc) [inline js/testimonial.js:13]: Why the comma is sometimes at the beginning of the line but sometimes at the end? Be consistent.

Comment (damithc) [inline js/testimonial.js:13]: I think the issue page lists 19 testimonials. Did you include all of them here?

File: css/main.less [hunk 9]
```diff
@@ -247,6 +247,15 @@ a, .btn-link, .accent {
   font-size: 16pt;
 }
 
+#testimonial-row{ 
+  width:760px; 
+  height:60px;
+  
+  font-size: 13pt;
+  font-family: Georgia, serif;
+  color:#A4A4A4
```
Comment (damithc) [inline css/main.less:256]: semicolon missing?

File: css/main.less [hunk 10]
```diff
@@ -247,6 +247,15 @@ a, .btn-link, .accent {
   font-size: 16pt;
 }
 
+#testimonial-row{
```
Comment (damithc) [inline css/main.less:250]: space missing (before brace)

File: css/main.less [hunk 11]
```diff
@@ -271,4 +280,6 @@ blockquote:after {
 blockquote p {
   margin-left: 3em;
   margin-right: 1em;
-}
\ No newline at end of file
+}
+
+
```
Comment (damithc) [inline css/main.less:285]: Unnecessary change?

File: index.html [hunk 12]
```diff
@@ -25,6 +25,20 @@
...
+			</div>
+			<div>
+				<img src="./img/quote-close.png" align="right"/>
+			</div>
+			<br />
+			<div>
+				<p  id="testimonial-row">
+				</p>
+			</div>
+		</div>
```
Comment (damithc) [inline index.html:41]: In future, we should standardize 2 spaces as indentation for html files (because they contain lot of nesting levels).

File: js/testimonial.js (hunk 8 above)
Comment (damithc) [inline js/testimonial.js:13]: See if you can limit line length in code files to about 110 chars. Otherwise it is hard to review (need to scroll horizontally)

**Comment by LIUJIAHAOCS** [comment]
In the issue page, some of the testimonials are too short, KAI suggest to take only the longer ones. Do i need to include them all?

**Comment by damithc** [comment]
I think we should include as many as we have. When it comes to testimonials, the quantity matters. Furthermore, sometimes the shorter ones can have more impact and be more memorable than the longer ones. However, you may want to ensure the the space taken by the testimonials remains the same. e.g. add extra line breaks for shorter ones?

**Comment by LIUJIAHAOCS** [comment]
New PR created, all testimonials are included.
